### PR TITLE
Add enhancements to UG and DG

### PR DESCRIPTION
**Pull request by JunCheng98**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ The `UI` component,
 
 1. `Logic` uses the `AddressBookParser` class to parse the user command.
 1. This results in a `Command` object which is executed by the `LogicManager`.
-1. The command execution can affect the `Model` (e.g. adding a person).
+1. The command execution can affect the `Model` (e.g. adding a animal).
```
Comment (lamlaaaam) [inline docs/DeveloperGuide.md]: "an animal"

Comment (JunCheng98) [reply docs/DeveloperGuide.md]: Fixed

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -724,29 +724,47 @@ testers are expected to do more *exploratory* testing.
...
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+   1. Test case: `delete 123`<br>
+      Expected: The animal with an ID of 123 deleted from the list. Details of the deleted animal shown in the status message. Timestamp in the status bar is updated.
 
    1. Test case: `delete 0`<br>
-      Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+      Expected: No animal is deleted. Error details shown in the status message. Status bar remains the same.
 
-   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is smaller than the requisite size)<br>
```
Comment (m0nggh) [inline docs/DeveloperGuide.md]: Think this one is for the delete index?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -724,29 +724,47 @@ testers are expected to do more *exploratory* testing.
...
+      2 animals are listed: an anaconda named 455 and a lion named Simba with an ID of 455.
+   
+   1. Test case: `delete 455`<br>
+      Expected: Simba with an ID of 455 is deleted from the list. Details of the deleted animal shown in the status message. Timestamp in the status bar is updated.
+
+   1. Test case: `delete 123`<br>
+      Expected: No animal is deleted. Error details shown in the status message. Status bar remains the same.
+
+   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is smaller than the requisite size)<br>
+      Expected: Similar to previous.
```
Comment (m0nggh) [inline docs/DeveloperGuide.md]: May have to modify this since delete should work without finding anymore.

Comment (JunCheng98) [reply docs/DeveloperGuide.md]: Deleted

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (JunCheng98) [reply docs/DeveloperGuide.md]: Updated it to 'where x is smaller than all 3 digit numbers' since these are examples of incorrect delete commands

**Pull request merged**